lib/services/computer.ts

```diff
@@ -0,0 +1,52 @@
+import db, { sql }  from "../db"
+import _            from "lodash"
+
+/**
+ * Returns the number of unread messages of a user for a given conversation
+ *
+ * @export
+ * @param {number} staffId
+ * @param {number} conversationId
+ * @returns {Promise<number>}
+ */
+export async function computeUnreadMessageCount(staffId: number, conversationId: number) : Promise<number> {
+  const [{ count }] = await sql`
+    WITH last_read_timestamps AS (
+      SELECT rr."conversationId", m."createdAt" AS timestamp
+        FROM "ReadReceipt" rr
+        JOIN "Message" m ON m.id = rr."lastReadMessageId"
+        WHERE rr."userType" = 'STAFF' AND rr."userId" = ${staffId}
+    )
+    SELECT count(m.id) FROM "Message" m
+      LEFT JOIN "ReadReceipt" rr ON (
+        rr."conversationId" = ${conversationId} AND
+        rr."userType" = 'STAFF' AND
+        rr."userId" = ${staffId}
+      )
+      LEFT JOIN last_read_timestamps AS lrt ON lrt."conversationId" = rr."conversationId"
+    WHERE m."conversationId" = ${conversationId}
+      AND (m."createdAt" > lrt.timestamp OR lrt.timestamp IS NULL)
+  `
```

Comment:
Any chance for SQL injection?

Reply:
it's uses a js template string, so prisma will escape all props